ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -347,14 +348,45 @@ impl<'a> Assemble<'a> {
                 }
                 dynasm!(self.asm; mov rcx, rsp);
 
+                // Push [GuardIdx]'s of previous guard failures.
+                if let Some(ref guards) = prevguards {
+                    for gidx in guards.iter().rev() {
+                        dynasm!(self.asm
+                            ; mov r9, QWORD (*gidx).into()
+                            ; push r9
+                        );
+                    }
+                }
+                // Save the pointer to this list.
+                dynasm!(self.asm
+                    ; mov r8, rsp
+                );
+
+                let len = i64::try_from(prevguards.as_ref().map_or(0, |v| v.len())).unwrap();
+                // Total alignment caused by pushing the parent guards.
+                let mut totalalign = len * 8;
+
+                // Pushing RSI above mis-aligned the stack. Unless we accidentally re-aligned it by
```

Comment:
How is it misaligned? Aren't the guard IDs 64 bits?

Reply:
We need to be 16 byte aligned before calls and pushing RDI further above misaligns the stack.

Reply:
Ah! Can we clarify that it's 16 bytes please (I incorrectly assumed 8).

Reply:
[ffda13e](https://github.com/ykjit/yk/pull/1461/commits/ffda13e4e44b0333b166d6da8a1e0e7afa6d12a1)